<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>415</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>415</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1024</width>
    <height>415</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Pensrose - 2D Polygon Transforms</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>penrose/img/icon.png</normaloff>penrose/img/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>361</width>
      <height>374</height>
     </rect>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_vertices">
     <attribute name="title">
      <string>Polygon Vertices</string>
     </attribute>
     <widget class="QPushButton" name="pushVertice">
      <property name="geometry">
       <rect>
        <x>280</x>
        <y>312</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string> Push</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>penrose/img/plus.png</normaloff>penrose/img/plus.png</iconset>
      </property>
      <property name="shortcut">
       <string>Ins</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="popVertice">
      <property name="geometry">
       <rect>
        <x>215</x>
        <y>312</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string> Pop</string>
      </property>
      <property name="icon">
       <iconset>
        <normaloff>penrose/img/cancel.png</normaloff>penrose/img/cancel.png</iconset>
      </property>
      <property name="shortcut">
       <string>Del</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QTableView" name="tableVertices">
      <property name="geometry">
       <rect>
        <x>-3</x>
        <y>-1</y>
        <width>361</width>
        <height>345</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="gridStyle">
       <enum>Qt::DashLine</enum>
      </property>
     </widget>
     <widget class="QTextEdit" name="helpText">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>311</width>
        <height>141</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::NoFrame</enum>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Cantarell'; font-size:10pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;You must insert at least &lt;span style=&quot; font-weight:600;&quot;&gt;3 vertices&lt;/span&gt; through the '&lt;span style=&quot; font-weight:600;&quot;&gt;push&lt;/span&gt;' button. After this, you can render your polygon clicking in &lt;span style=&quot; font-weight:600;&quot;&gt;Tools &amp;gt; Render&lt;/span&gt;, or pressing &lt;span style=&quot; font-weight:600;&quot;&gt;F4 key&lt;/span&gt; shortcut. &lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;If you need help, go to the &lt;/span&gt;&lt;span style=&quot; font-weight:600; font-style:italic;&quot;&gt;Help &amp;gt; Github repository&lt;/span&gt;&lt;span style=&quot; font-style:italic;&quot;&gt; to read the documentation.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
     <zorder>tableVertices</zorder>
     <zorder>popVertice</zorder>
     <zorder>pushVertice</zorder>
     <zorder>helpText</zorder>
    </widget>
    <widget class="QWidget" name="tab_transformations">
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <attribute name="title">
      <string>2D Transformations</string>
     </attribute>
     <widget class="QGroupBox" name="boxTranslation">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>161</width>
        <height>95</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Translation</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QDoubleSpinBox" name="translation_tx">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="wrapping">
        <bool>false</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="translation_ty">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Tx:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Ty:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="boxScale">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>161</width>
        <height>121</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Scale</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QDoubleSpinBox" name="scale_sx">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="scale_sy">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Sx:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_4">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Sy:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Vertices:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QComboBox" name="scale_point">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>90</y>
         <width>81</width>
         <height>25</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="boxShearing">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>10</y>
        <width>161</width>
        <height>95</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Shear</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QDoubleSpinBox" name="shear_cx">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="wrapping">
        <bool>false</bool>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="decimals">
        <number>2</number>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QDoubleSpinBox" name="shear_cy">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::PlusMinus</enum>
       </property>
       <property name="minimum">
        <double>-100.000000000000000</double>
       </property>
       <property name="maximum">
        <double>100.000000000000000</double>
       </property>
      </widget>
      <widget class="QLabel" name="label_6">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>30</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Cx:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_7">
       <property name="geometry">
        <rect>
         <x>40</x>
         <y>60</y>
         <width>21</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Cy:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="boxRotation">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>110</y>
        <width>161</width>
        <height>121</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Rotation</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>41</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Angle:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>90</y>
         <width>51</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Vertices:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QComboBox" name="rotation_vertices">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>90</y>
         <width>81</width>
         <height>25</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QComboBox" name="rotation_direction">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>60</y>
         <width>81</width>
         <height>25</height>
        </rect>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>81</width>
         <height>25</height>
        </size>
       </property>
       <property name="editable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>60</y>
         <width>61</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Direction:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
      <widget class="QSpinBox" name="rotation_angle">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>30</y>
         <width>81</width>
         <height>26</height>
        </rect>
       </property>
       <property name="buttonSymbols">
        <enum>QAbstractSpinBox::UpDownArrows</enum>
       </property>
       <property name="specialValueText">
        <string/>
       </property>
       <property name="showGroupSeparator" stdset="0">
        <bool>false</bool>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>360</number>
       </property>
       <property name="singleStep">
        <number>45</number>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="boxReflection">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>237</y>
        <width>151</width>
        <height>95</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Reflection</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <widget class="QCheckBox" name="reflection_x">
       <property name="geometry">
        <rect>
         <x>70</x>
         <y>47</y>
         <width>31</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>X</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="reflection_y">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>47</y>
         <width>31</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Y</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>47</y>
         <width>31</width>
         <height>20</height>
        </rect>
       </property>
       <property name="text">
        <string>Axes:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QFrame" name="outputFrame">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>37</y>
      <width>634</width>
      <height>346</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">background:white</string>
    </property>
    <property name="frameShape">
     <enum>QFrame::NoFrame</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QLabel" name="label_14">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>7</y>
       <width>56</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>output:</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionOpen_file"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionGithub_repository"/>
    <addaction name="actionAbout_this_software"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionCompile"/>
    <addaction name="actionUpdate"/>
    <addaction name="separator"/>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionClear">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/new.png</normaloff>penrose/img/new.png</iconset>
   </property>
   <property name="text">
    <string>Clear data</string>
   </property>
   <property name="toolTip">
    <string>Clears all entered data</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen_file">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/open.png</normaloff>penrose/img/open.png</iconset>
   </property>
   <property name="text">
    <string>Open file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/save.png</normaloff>penrose/img/save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/close.png</normaloff>penrose/img/close.png</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionGithub_repository">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/github.png</normaloff>penrose/img/github.png</iconset>
   </property>
   <property name="text">
    <string>Github repository</string>
   </property>
  </action>
  <action name="actionAbout_this_software">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/doc.png</normaloff>penrose/img/doc.png</iconset>
   </property>
   <property name="text">
    <string>About</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionCompile">
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/settings.png</normaloff>penrose/img/settings.png</iconset>
   </property>
   <property name="text">
    <string>Render</string>
   </property>
   <property name="shortcut">
    <string>F4</string>
   </property>
  </action>
  <action name="actionUpdate">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>penrose/img/update.png</normaloff>penrose/img/update.png</iconset>
   </property>
   <property name="text">
    <string>Update</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
